src/vm/methodtablebuilder.cpp

```diff
@@ -1501,7 +1501,7 @@ MethodTableBuilder::BuildMethodTableThrowing(
         if (hr == S_OK && strcmp(nameSpace, "System.Runtime.Intrinsics.Arm.Arm64") == 0)
 #else
         // All the funtions in System.Runtime.Intrinsics.X86 are hardware intrinsics.
-        if (hr == S_OK && strcmp(nameSpace, "System.Runtime.Intrinsics.X86") == 0)
+        if (hr == S_OK && ((strcmp(nameSpace, "System.Runtime.Intrinsics.X86") == 0) || (bmtInternal->pType->IsNested() && strcmp(className, "X64") == 0)))
```

Comment:
For the x64 case shouldn't this also (or instead) check the enclosing class namespace?

Reply:
This code is guarded by `GetModule()->IsSystem()`, and we only have hardware intrinsic X64 classes in mscorlib now. So, probably adding an assert for enclosing class namespace is enough?